### PR TITLE
Disclosure consents and external parties

### PR DESCRIPTION
**Pull request by dipeshfrollo**
[WA-4001] and [WA-4003]

Checklist

- Is the destination branch correct & intended? ✅
- Is the correct JIRA ticket referenced? ✅
- Is the correct API docs/Confluence Page referenced?
- Have all the AC's in the ticket passed? ✅
- Have you followed proper naming conventions in the code? ✅
- Has the code been properly covered with unit tests? ✅
- Have you regenerated the mock DBs, if the datamodel was modified ✅
- Have you added migration tests (if applicable)? ✅
- Have the unit tests passed locally (if applicable)? ✅
- Have you addressed the code warnings if any?
- Have you tested the API and made sure the documentation is up-to-date?
- Have you added documentation for all public methods & types ✅

[WA-4001]: https://frollo.atlassian.net/browse/WA-4001?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WA-4003]: https://frollo.atlassian.net/browse/WA-4003?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by NeilCossorFrollo — APPROVED** [review]
All good **except** think you have a typo has snuck in.. pls correct the name of "DisclosureConsent +CoreDataProperties.swift" to get rid of the space.. which are a pain on MacOS at cmdline

**Comment by dipeshfrollo** [comment]
> 

Thanks, eagle eye :D . I have updated the PR with new commit (commit name has a typo LOL)

File: Sources/FrolloSDK/Aggregation/Aggregation+CDR.swift [hunk 1]
```diff
@@ -0,0 +1,851 @@
...
+                        predicates.append(NSPredicate(format: #keyPath(DisclosureConsent.consentID) + " > %ld",
+                                                      argumentArray: [beforeID]))
+                    }
+
+                    if let afterIDString = response.paging?.cursors?.after, let afterID = Int64(afterIDString) {
+                        predicates.append(NSPredicate(format: #keyPath(DisclosureConsent.consentID) + " <= %ld",
+                                                      argumentArray: [afterID]))
+                    }
+
+                    self.handleDisclosureConsentResponse(response.data.elements,
```
Comment (sincychacko-frollo) [inline Sources/FrolloSDK/Aggregation/Aggregation+CDR.swift:415]: Should we also link collection consents?
And add a convenience method to fetch from cache>

Comment (dipeshfrollo) [reply Sources/FrolloSDK/Aggregation/Aggregation+CDR.swift:415]: It is a Many to Many?

**Review by sincychacko-frollo — APPROVED** [review]

**Pull request merged**